### PR TITLE
Draft an order prod fix

### PR DESCRIPTION
**Pull request by BharadwajSairam**
### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java [hunk 1]
```diff
@@ -196,13 +173,13 @@ public AboutToStartOrSubmitCallbackResponse populateJudgeOrAdminDraftOrderCustom
             DraftOrder selectedOrder = draftAnOrderService.getSelectedDraftOrderDetails(caseData);
             if (selectedOrder != null && (CreateSelectOrderOptionsEnum.blankOrderOrDirections.equals(selectedOrder.getOrderType()))
             ) {
-                caseData = draftAnOrderService.generateDocument(callbackRequest, caseData);
+                caseData = draftAnOrderService.updateCustomFieldsWithSdoAoGuardianForMorders(callbackRequest, caseData);
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/EditAndApproveDraftOrderController.java:176]: Method name can be changed?

File: src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java [hunk 2]
```diff
@@ -225,10 +235,8 @@ public Map<String, Object> removeDraftOrderAndAddToFinalOrder(String authorisati
                 } else {
                     draftOrder = getDraftOrderWithUpdatedStatus(caseData, eventId, loggedInUserType, draftOrder);
                 }
-                updatedCaseData.put(
-                    "orderCollection",
-                    getFinalOrderCollection(authorisation, caseData, draftOrder, eventId)
-                );
+                List<Element<OrderDetails>> orderCollection = getFinalOrderCollection(authorisation, caseData, draftOrder, eventId);
+                updatedCaseData.put("orderCollection", orderCollection);
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java:239]: is this change needed?

File: src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java [hunk 3]
```diff
@@ -567,11 +573,30 @@ public Map<String, Object> populateCommonDraftOrderFields(String authorization,
...
+        String caseReferenceNumber = String.valueOf(caseData.getId());
+        if (CollectionUtils.isEmpty(manageOrderHearingDetail)) {
+            Hearings hearings = hearingService.getHearings(authorization, caseReferenceNumber);
+            HearingDataPrePopulatedDynamicLists hearingDataPrePopulatedDynamicLists =
+                hearingDataService.populateHearingDynamicLists(authorization, caseReferenceNumber, caseData, hearings);
+            HearingData hearingData = hearingDataService.generateHearingData(
+                hearingDataPrePopulatedDynamicLists, caseData);
+            manageOrderHearingDetail = ElementUtils.wrapElements(hearingData);
+        }
+        if (!Event.DRAFT_AN_ORDER.getId().equalsIgnoreCase(caseEventId) && Yes.equals(orderDraftedBySolicitor)) {
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java:594]: First part of if condition will it ever happens?

File: src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java [hunk 4]
```diff
@@ -604,6 +629,16 @@ public Map<String, Object> updateDraftOrderCollection(CaseData caseData, String
             if (e.getId().equals(selectedOrderId)) {
                 if (YesOrNo.Yes.equals(caseData.getDoYouWantToEditTheOrder()) || (caseData.getManageOrders() != null
                     && Yes.equals(caseData.getManageOrders().getMakeChangesToUploadedOrder()))) {
+                    if (caseData.getManageOrders().getOrdersHearingDetails() != null
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java:632]: Even for uploaded orders, do we need to set hearing details?

File: src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java [hunk 5]
```diff
@@ -1191,33 +1228,35 @@ public Map<String, Object> getDraftOrderInfo(String authorisation, CaseData case
...
         if (Event.ADMIN_EDIT_AND_APPROVE_ORDER.getId().equalsIgnoreCase(eventId)
             && (WhatToDoWithOrderEnum.finalizeSaveToServeLater
             .equals(caseData.getServeOrderData().getWhatDoWithOrder())
             || YesOrNo.Yes.equals(caseData.getServeOrderData().getDoYouWantToServeOrder()))) {
-            CaseData updatedCaseData = objectMapper.convertValue(
-                caseDataUpdated,
-                CaseData.class
-            );
-            caseDataUpdated.putAll(removeDraftOrderAndAddToFinalOrder(authorisation, updatedCaseData, eventId));
+            caseDataUpdated.putAll(removeDraftOrderAndAddToFinalOrder(authorisation, caseData, eventId));
```
Comment (shashisk97) [inline src/main/java/uk/gov/hmcts/reform/prl/services/DraftAnOrderService.java:1250]: I dont remember exactly but there was a valid reason to have a clone of caseData as updatedCaseData, is it not needed?

**Review by shashisk97 — COMMENTED** [review]
Loggers needs to be removed after QA

**Review by yogendraU — APPROVED** [review]

**Review by sahaSonali — APPROVED** [review]
